agent/app/agent.go

```diff
@@ -660,3 +661,19 @@ func mergeTags(localTags []*ecs.Tag, ec2Tags []*ecs.Tag) []*ecs.Tag {
 
 	return utils.MapToTags(tagsMap)
 }
+
+// getHostPublicIPv4AddressFromEC2API will retrieve the PublicIPAddress (IPv4) of this
+// instance through the EC2 API
+func (agent *ecsAgent) getHostPublicIPv4AddressFromEC2API() string {
+	// Get instance ID from ec2 metadata client.
+	instanceID, err := agent.ec2MetadataClient.InstanceID()
+	if err != nil {
+		return ""
+	}
+
+	hostPublicIPv4Address, err := agent.ec2Client.GetHostPublicIPv4AddressForInstance(instanceID)
```

Comment:
You can get this from the EC2 instance metadata, without making a call to the EC2 API (and requiring increased IAM permissions):

```bash
$ curl 169.254.169.254/latest/meta-data/public-ipv4
```

Reply:
In the GitHub issue, the request was to put it in the metadata file as an alternative to using the EC2 instance metadata. I think the use case being when the EC2 instance metadata service is disabled, they'd still like to see the HostIP 

> "This is a security risk as enabling instance meta data allows for querying IAM credentials as well."

Reply:
It's not possible to disable the EC2 instance metadata service on the host, but it is possible to block container access to the IMDS using iptables rules.  The ECS agent has a facility for doing this when tasks use the `awsvpc` network mode with the `ECS_AWSVPC_BLOCK_IMDS` setting.

Note that the ECS agent relies on the IMDS being available for a variety of purposes, so you cannot have a functioning ECS agent without having access to IMDS.  (Your code above even has a dependency on it in this line: `instanceID, err := agent.ec2MetadataClient.InstanceID()`.  You should be able to change this to grab the public IP address instead of the instance ID and skip the rest of the logic you have.)